### PR TITLE
Groups pages settings

### PR DESCRIPTION
**Pull request by RPHaynes**

### CODE REVIEW

**Review by LutherNa — CHANGES_REQUESTED** [review]
fix merge conflicts. and if you mess up the formatting Jennica will be rather cross.

**Review by LutherNa — CHANGES_REQUESTED** [review]
Tested with backend branch "main", frontend branch "groups-pages-settings" with main merged. Some of the issues below may be due to bad merging, others are more likely native to the groups-pages-settings branch.
- Saving user page settings on a user page with nested posts causes an error on the backend due to saving the same post entity multiple times in the same transaction. 
- On the FriendsTab element, mapped friends do not have unique keys.
- FriendsTab has console.log() for debugging still enabled (_this should always be removed for merges to main_). 
- Adding a friend and joining a group return a 404 error, as those endpoints do not yet exist on main (and there are no pull requests open for backend changes, I checked). 
- Saving group page settings (even if there are no changes) causes the Settings button to turn into the Options button until I refresh the page. This also produces the following console error: 
`
MUI: The `value` provided to the Tabs component is invalid.
None of the Tabs' children match with "3".
You can provide one of the following values: 0, 1, 2. 
`
- Because the Join Chat button is on a dropdown off the Options button, I cannot attempt to join a chatroom for a page I own.
- Chat button doesn't seem to do anything, but I assume that's a WIP. 
- FriendsTab throwing the following error: 
`
react_devtools_backend.js:4045 Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
`
- Please put all API queries in the API folder.

**Comment by RPHaynes** [comment]
Merge conflicts fixed

**Comment by RPHaynes** [comment]
Backend Pull request Opened

**Review by Justin-Sherfey — APPROVED** [review]

File: src/Components/Page/FriendsTab.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import { useEffect, useState } from "react";
+import APIQuery from "../../API/APIQuery";
+import ListItem from "@mui/material/ListItem";
+import ListItemButton from "@mui/material/ListItemButton";
+import { Box, ListItemText } from "@mui/material";
+import ListItemIcon from "@mui/material/ListItemIcon";
+import InboxIcon from "@mui/icons-material/Inbox";
+import { Navigate, useNavigate, useParams } from "react-router-dom";
+import { createBrowserHistory } from "history";
+
```
Comment (JennicaLeClerc) [inline src/Components/Page/FriendsTab.js:10]: Comment

File: src/Components/Group/CreateGroup.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    Paper,
+    Tab,
+    Tabs,
+    TextField,
+    Tooltip,
+    Typography
+} from "@mui/material";
+import APIQuery from "../../API/APIQuery";
+import { useState } from "react";
+
```
Comment (JennicaLeClerc) [inline src/Components/Group/CreateGroup.js:26]: Comment

File: src/Components/Group/AddMember.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import {
+    Button,
+    Typography,
+    TextField,
+} from "@mui/material";
+import APIQuery from "../../API/APIQuery";
+import { useState } from "react";
+
```
Comment (JennicaLeClerc) [inline src/Components/Group/AddMember.js:8]: Comment

File: src/Components/Page/PageSetting.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+	FormGroup,
+	TextField,
+	Switch,
+	Grid,
+} from "@mui/material";
+
+import APIQuery from "../../API/APIQuery";
+import { useState } from "react";
+import { Box } from "@mui/system";
+
```
Comment (JennicaLeClerc) [inline src/Components/Page/PageSetting.js:12]: Comment

**Review by JennicaLeClerc — COMMENTED** [review]
COMMENTS!!!!!

**Review by LutherNa — APPROVED** [review]

**Review by AaronLitton — APPROVED** [review]
Untested, but looks good to go

**Review by JennicaLeClerc — APPROVED** [review]
thanks

**Pull request merged**